Add a project item

cdverrett94 opened the pull request:
Closes #248

- Adds a project item and its embed
   - If the initial creation data includes the item UUID, all of the item's project info is transferred to the project item and the name will be updated to "Craft {item name}"
   - The interactivity with characters will happen when it gets added to the character sheet as part of #249 
- Moved the equipment's item prerequisite to the project schema since the prereq is project related. If you need me to migrate this, I might need some help because I'm not sure how to do migrations.

![project-item](https://github.com/user-attachments/assets/24254fbc-3f62-4c8b-955b-172e17184a3c)

JPMeehan commented:
We're early enough I'm not terribly concerned about providing comprehensive migrations.

cdverrett94 commented:
Alright, power is finally back on, so was able to make the changes per request. Also updated the parameters of the goal/progress fields and added a progress input to the project sheet when its owned.
